$OpenBSD: patch-taglib_mpeg_id3v2_id3v2framefactory_cpp,v 1.1 2012/11/07 09:48:20 dcoppa Exp $

Fix noop frame id size check in ID3v2::FrameFactory::createFrame
(upstream git commit c42bdeab43dc1a0fa1c15ddf025e76932ba26f62)

--- taglib/mpeg/id3v2/id3v2framefactory.cpp.orig	Wed Nov  7 10:19:03 2012
+++ taglib/mpeg/id3v2/id3v2framefactory.cpp	Wed Nov  7 10:19:49 2012
@@ -99,7 +99,7 @@ Frame *FrameFactory::createFrame(const ByteVector &ori
   // A quick sanity check -- make sure that the frameID is 4 uppercase Latin1
   // characters.  Also make sure that there is data in the frame.
 
-  if(!frameID.size() == (version < 3 ? 3 : 4) ||
+  if(frameID.size() != (version < 3 ? 3 : 4) ||
      header->frameSize() <= uint(header->dataLengthIndicator() ? 4 : 0) ||
      header->frameSize() > data.size())
   {
